app/src/main/java/org/schabi/newpipe/player/resolver/PlaybackResolver.java

```diff
@@ -68,38 +87,198 @@ default MediaSource buildLiveMediaSource(@NonNull final PlayerDataSource dataSou
 
     @NonNull
     default MediaSource buildMediaSource(@NonNull final PlayerDataSource dataSource,
```

Comment:
Remove `default` from all of these methods and make them `static` instead:
- they are methods of **an interface**, so they can't (and don't) do anything with the object on which they are called (since an interface has no fields / data)
- `default` methods are meant to be overridden, but none of these methods are overridden anywhere
- these methods do not do anything with `this`, i.e. they are already "static"

Reply:
This is the diff to apply, an it works:
```diff
diff --git a/app/src/main/java/org/schabi/newpipe/player/resolver/AudioPlaybackResolver.java b/app/src/main/java/org/schabi/newpipe/player/resolver/AudioPlaybackResolver.java
index e92a2722f..3be6503b9 100644
--- a/app/src/main/java/org/schabi/newpipe/player/resolver/AudioPlaybackResolver.java
+++ b/app/src/main/java/org/schabi/newpipe/player/resolver/AudioPlaybackResolver.java
@@ -37,7 +37,7 @@ public class AudioPlaybackResolver implements PlaybackResolver {
     @Override
     @Nullable
     public MediaSource resolve(@NonNull final StreamInfo info) {
-        final MediaSource liveSource = maybeBuildLiveMediaSource(dataSource, info);
+        final MediaSource liveSource = PlaybackResolver.maybeBuildLiveMediaSource(dataSource, info);
         if (liveSource != null) {
             return liveSource;
         }
@@ -54,8 +54,8 @@ public class AudioPlaybackResolver implements PlaybackResolver {
         final MediaSourceTag tag = new MediaSourceTag(info);
 
         try {
-            return buildMediaSource(dataSource, audio, info, PlayerHelper.cacheKeyOf(info, audio),
-                    tag);
+            return PlaybackResolver.buildMediaSource(
+                    dataSource, audio, info, PlayerHelper.cacheKeyOf(info, audio), tag);
         } catch (final IOException e) {
             Log.e(TAG, "Unable to create audio source:", e);
             return null;
diff --git a/app/src/main/java/org/schabi/newpipe/player/resolver/PlaybackResolver.java b/app/src/main/java/org/schabi/newpipe/player/resolver/PlaybackResolver.java
index 17cb2bbee..29af4585a 100644
--- a/app/src/main/java/org/schabi/newpipe/player/resolver/PlaybackResolver.java
+++ b/app/src/main/java/org/schabi/newpipe/player/resolver/PlaybackResolver.java
@@ -39,7 +39,7 @@ import java.util.Objects;
 public interface PlaybackResolver extends Resolver<StreamInfo, MediaSource> {
 
     @Nullable
-    default MediaSource maybeBuildLiveMediaSource(@NonNull final PlayerDataSource dataSource,
+    static MediaSource maybeBuildLiveMediaSource(@NonNull final PlayerDataSource dataSource,
                                                   @NonNull final StreamInfo info) {
         final StreamType streamType = info.getStreamType();
         if (!StreamTypeUtil.isLiveStream(streamType)) {
@@ -57,7 +57,7 @@ public interface PlaybackResolver extends Resolver<StreamInfo, MediaSource> {
     }
 
     @NonNull
-    default MediaSource buildLiveMediaSource(@NonNull final PlayerDataSource dataSource,
+    static MediaSource buildLiveMediaSource(@NonNull final PlayerDataSource dataSource,
                                              @NonNull final String sourceUrl,
                                              @C.ContentType final int type,
                                              @NonNull final MediaSourceTag metadata) {
@@ -86,7 +86,7 @@ public interface PlaybackResolver extends Resolver<StreamInfo, MediaSource> {
     }
 
     @NonNull
-    default MediaSource buildMediaSource(@NonNull final PlayerDataSource dataSource,
+    static MediaSource buildMediaSource(@NonNull final PlayerDataSource dataSource,
                                          @NonNull final Stream stream,
                                          @NonNull final StreamInfo streamInfo,
                                          @NonNull final String cacheKey,
@@ -104,7 +104,7 @@ public interface PlaybackResolver extends Resolver<StreamInfo, MediaSource> {
         }
     }
 
-    default <T extends Stream> ProgressiveMediaSource buildProgressiveMediaSource(
+    static <T extends Stream> ProgressiveMediaSource buildProgressiveMediaSource(
             @NonNull final PlayerDataSource dataSource,
             @NonNull final T stream,
             @NonNull final String cacheKey,
@@ -125,7 +125,7 @@ public interface PlaybackResolver extends Resolver<StreamInfo, MediaSource> {
         }
     }
 
-    default <T extends Stream> DashMediaSource buildDashMediaSource(
+    static <T extends Stream> DashMediaSource buildDashMediaSource(
             @NonNull final PlayerDataSource dataSource,
             @NonNull final T stream,
             @NonNull final StreamInfo streamInfo,
@@ -156,7 +156,7 @@ public interface PlaybackResolver extends Resolver<StreamInfo, MediaSource> {
         }
     }
 
-    default <T extends Stream> String generateDashManifestOfYoutubeDashStream(
+    static <T extends Stream> String generateDashManifestOfYoutubeDashStream(
             @NonNull final StreamInfo streamInfo,
             @NonNull final T stream) throws IOException {
         final String content;
@@ -209,7 +209,7 @@ public interface PlaybackResolver extends Resolver<StreamInfo, MediaSource> {
         return content;
     }
 
-    default <T extends Stream> DashMediaSource createDashMediaSource(
+    static <T extends Stream> DashMediaSource createDashMediaSource(
             @NonNull final PlayerDataSource dataSource,
             @NonNull final String content,
             @NonNull final T stream,
@@ -238,7 +238,7 @@ public interface PlaybackResolver extends Resolver<StreamInfo, MediaSource> {
                         .build());
     }
 
-    default <T extends Stream> HlsMediaSource buildHlsMediaSource(
+    static <T extends Stream> HlsMediaSource buildHlsMediaSource(
             @NonNull final PlayerDataSource dataSource,
             @NonNull final T stream,
             @NonNull final String cacheKey,
diff --git a/app/src/main/java/org/schabi/newpipe/player/resolver/VideoPlaybackResolver.java b/app/src/main/java/org/schabi/newpipe/player/resolver/VideoPlaybackResolver.java
index 598548242..ec1cbd9b0 100644
--- a/app/src/main/java/org/schabi/newpipe/player/resolver/VideoPlaybackResolver.java
+++ b/app/src/main/java/org/schabi/newpipe/player/resolver/VideoPlaybackResolver.java
@@ -52,7 +52,7 @@ public class VideoPlaybackResolver implements PlaybackResolver {
     @Override
     @Nullable
     public MediaSource resolve(@NonNull final StreamInfo info) {
-        final MediaSource liveSource = maybeBuildLiveMediaSource(dataSource, info);
+        final MediaSource liveSource = PlaybackResolver.maybeBuildLiveMediaSource(dataSource, info);
         if (liveSource != null) {
             return liveSource;
         }
@@ -80,8 +80,8 @@ public class VideoPlaybackResolver implements PlaybackResolver {
 
         if (video != null) {
             try {
-                final MediaSource streamSource = buildMediaSource(dataSource, video, info,
-                        PlayerHelper.cacheKeyOf(info, video), tag);
+                final MediaSource streamSource = PlaybackResolver.buildMediaSource(
+                        dataSource, video, info, PlayerHelper.cacheKeyOf(info, video), tag);
                 mediaSources.add(streamSource);
             } catch (final IOException e) {
                 Log.e(TAG, "Unable to create video source:", e);
@@ -99,8 +99,8 @@ public class VideoPlaybackResolver implements PlaybackResolver {
         // merge with audio stream in case if video does not contain audio
         if (audio != null && (video == null || video.isVideoOnly())) {
             try {
-                final MediaSource audioSource = buildMediaSource(dataSource, audio, info,
-                        PlayerHelper.cacheKeyOf(info, audio), tag);
+                final MediaSource audioSource = PlaybackResolver.buildMediaSource(
+                        dataSource, audio, info, PlayerHelper.cacheKeyOf(info, audio), tag);
                 mediaSources.add(audioSource);
             } catch (final IOException e) {
                 Log.e(TAG, "Unable to create audio source:", e);
```

Reply:
Thank you! You are sometimes crazy, Stypox!